whitepapers/0011_accountant.md

```diff
@@ -0,0 +1,190 @@
+    },
+
+    /// Returns the number of signatures necessary for quorum for the given guardian set index.
+    #[returns(u32)]
+    CalculateQuorum { guardian_set_index: u32 },
+}
+```
+
+#### Submitting observations to wormchain
+
+When a guardian observes a token transfer message from the registered tokenbridge on a particular chain, it will submit that observation directly to the accountant.  The contract will keep track of pending token transfers and emit an event once it has received a quorum of observations.  If a quorum of guardians submit their observations within a single tendermint block then the token transfer only needs to wait for one round of tendermint consensus and one round of guardian consensus.
+
+When a guardian submits an observation to the contract, the contract will perform several checks:
+
+- Check that the signature is valid and from a guardian in the current guardian set.
+- If a transfer with the same `(emitter_chain, emitter_address, sequence)` tuple has already been committed, then check that the digest of the observation matches the digest of the committed transfer.  If they match return a `Committed` status; otherwise return an `Error` status and emit an `Error` event.
+- If the transfer has not been committed, mark the observation as having a signature from the guardian.
+- If the observation does not yet have a quorum of signatures, returning a `Pending` status.
+- If the observation has a quorum of signatures, parse the tokenbridge payload and commit the transfer.
+- If committing the transfer fails (for example, if one of the accounts does not have a sufficient balance) then return an `Error` status and emit an `Error` event.  Otherwise return a `Committed` status and emit a `Committed` event.
+
+If the guardian receives a `Committed` status for a transfer then it can immediately proceed to signing the VAA.  If it receives an `Error` status, then it should halt further processing of the transfer (see the [Handling Rejections](https://www.notion.so/tbjump-copy-Accounting-v3-11572f647daf46f1a1e7d3ac80050ac0) section for more details).  If it receives a `Pending` status, then it should add the observation to a local database of pending transfers and wait for an event from the accountant.
+
+#### Observing Events from Wormhole Chain
+
+In addition to submitting observations to the accountant, each guardian must also set up a watcher to watch for transfers that complete asynchronously, which can happen when no observation has reached quorum for a particular `(emitter_chain, emitter_address, seuence)` tuple.
+
+There are 2 types of events that the guardian must handle:
+
+- Transfer committed - This indicates that an observation has reached quorum and the transfer has been committed.  The guardian must independently verify the contents of the event and then sign the VAA if they match.  See [this section](https://www.notion.so/tbjump-copy-Accounting-v3-11572f647daf46f1a1e7d3ac80050ac0) for more details.
```

Comment:
what does independently verify mean here, notion link

Reply:
Fixed by @conorpp in e3f7a7d5d2ec5e7220c682b4a48b5e88eb40dd08.